website/content/docs/k8s/installation/vault/wan-federation.mdx

```diff
@@ -0,0 +1,678 @@
+
+    <Tabs>
+      <Tab heading="EKS">
+
+      ```shell-session
+      export MESH_GW_HOST=$(kubectl get svc consul-mesh-gateway -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+      ```
+
+      </Tab>
+
+      <Tab heading="GKE">
+
+      ```shell-session
+      export MESH_GW_HOST=$(kubectl get svc consul-mesh-gateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+      ```
+
+      </Tab>
+
+      <Tab heading="AKS">
+
+      ```shell-session
+      export MESH_GW_HOST=$(kubectl get svc consul-mesh-gateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+      ```
+
+      </Tab>
+    </Tabs>
+
+1. Configure and install Consul in the secondary datacenter (dc2).
+    
+    -> Note To configure Vault as the Connect CA in secondary datacenters, you need to make sure that the Root CA path is the same, but the intermediate is different for each datacenter. In the `connectCA` Helm configuration for a secondary datacenter, you can specify a `intermediatePKIPath` that is, for example, prefixed with the datacenter for which this configuration is intended (e.g. `dc2/connect-intermediate`).
+
+    <CodeBlockConfig filename="consul-dc2.yaml" linenumbers highlight="5,6,7,8,9,10,11,12,13,14,15,16,20,21,22,23,24,25,26,27,30,31,32,33,34,35,38,39">
+
+    ```yaml
+    # consul-dc2.yaml
+    global:
+      datacenter: "dc2"
+      name: consul
+      secretsBackend:
+        vault:
+          enabled: true
+          consulServerRole: consul-server
+          consulClientRole: consul-client
+          consulCARole: consul-ca
+          manageSystemACLsRole: server-acl-init
+          connectCA:
+            address: ${VAULT_ADDR}
+            rootPKIPath: connect_root/
+            intermediatePKIPath: dc2/connect_inter/
+            authMethodPath: kubernetes-dc2
+      tls:
+        enabled: true
+        enableAutoEncrypt: true
+        caCert:
+          secretName: "pki/cert/ca"
+      federation:
+        enabled: true
+        primaryDatacenter: dc1
+        k8sAuthMethodHost: ${KUBE_API_URL_DC2}
+        primaryGateways:
+        - ${MESH_GW_HOST}:443
+      acls:
+        manageSystemACLs: true
+        replicationToken:
+          secretName: consul/data/secret/replication
+          secretKey: token
+      gossipEncryption:
+        secretName: consul/data/secret/gossip
+        secretKey: key
+    server:
+      replicas: 1
+      serverCert:
+        secretName: "pki/issue/consul-cert-dc2"
+    connectInject:
+      replicas: 1
+      enabled: true
+    controller:
+      enabled: true
+    meshGateway:
+      enabled: true
+      replicas: 1
+    ```
+
+    </CodeBlockConfig>
+
+    Change your Kubernetes current context to target the secondary datacenter (dc2):
+
+    ```shell-session
+    $ kubectl config use-context <context for dc2>
+    ```
+
+    Now, use the above values files for you helm install:
+
+    ```shell-session
+    $ helm install consul-dc2 -f consul-dc2.yaml hashicorp/consul
+    ```
+
+## Next steps
```

Comment:
Is there a quick validation step we can do? How can the practitioner feel confident they set this up correctly?

Reply:
How is [this change](https://github.com/hashicorp/consul/pull/12802/commits/cc0c0c00dfcf0b4ffb183ba5c6fc549711237aa8)?  This is as far as we have taken this process.  We have not run any examples deploying hashicups or anything like that.  That might be more appropriate for a learn guide?

Reply:
It's better but still not ideal. But I understand the situation as no services are applied.

Reply:
I have demo showing this and to validate everything, I have run this command to confirm that WAN Fed is up and running:
`kubectl exec consul-server-0 --context=dc1 -- consul members -wan -ca-file /vault/secrets/serverca.crt`
Maybe that helps conclude that WAN fed was setup correctly?

Reply:
This is likely beyond the scope, but what I also do in my demo is look and compare the certs and ACL tokens stored in Vault and the certs/tokens used by the Consul servers. If they match, the confirms Consul is using correct certs generated by Vault.